Sources/MapboxDirectionsCLI/CodingOperation.swift

```diff
@@ -29,6 +32,36 @@ class CodingOperation<ResponceType : Codable, OptionsType : DirectionsOptions >
                let jsonData = try? JSONSerialization.data(withJSONObject: object, options: [.prettyPrinted]) {
                 outputText = String(data: jsonData, encoding: .utf8)!
             }
+            
+        case .gpx:
+            #if !os(Linux)
+            var gpxText: String = String("<?xml version=\"1.0\" encoding=\"UTF-8\"?>")
+            gpxText.append("\n<gpx xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns=\"http://www.topografix.com/GPX/1/1\" xsi:schemaLocation=\"http://www.topografix.com/GPX/1/1 http://www.topografix.com/GPX/1/1/gpx.xsd\" version=\"1.1\">")
+            
+            guard let routeResponse = routeResponse,
+                  let routes = routeResponse.routes else { return }
+            
+            let timeInterval: TimeInterval = 1
+            let dateFormatter = ISO8601DateFormatter()
+            dateFormatter.formatOptions = .withInternetDateTime
+            var time = Date()
+            
+            routes.forEach { route in
```

Comment:
Should we somehow split different routes within the GPX file? At least by adding an XML comment.

Reply:
A comment would be helpful for debugging purposes; otherwise, I think it’s OK that the track jumps from the end of one route to the beginning of the next: https://github.com/mapbox/mapbox-directions-swift/pull/572#discussion_r719564488.

Reply:
It's OK, sure. But I can imagine the case when it might be inconvenient for a user to make a different route request, but they might still want to simulate only for one of the routes. We can add a CLI argument to filter a route, but this comment idea is just a bit simpler.